pkg/test/e2e/defaultappcatalog/application_test.go

```diff
@@ -360,3 +365,31 @@ func createUserCluster(
 
 	return clusterClient, cleanup, log, err
 }
+
+// makeValuesBlockWithTemplating takes a string of values and returns a string in order to be used for templating.
+func makeValuesBlockWithTemplating(name string, values string) string {
+	var customLine string
+	if strings.Contains(defaultValuesBlock, "null") || defaultValuesBlock == "" {
+		defaultValuesBlock = ""
+		customLine = fmt.Sprintf("customVar: %s\n", customVarTemplate)
+	} else {
+		// Ensure proper newline separation if appending
+		customLine = fmt.Sprintf("\ncustomVar: %s\n", customVarTemplate)
+	}
+
+	if !isExcluded(name, excludeApplicationsFromTestingTemplating) {
+		return values + customLine
+	}
```

Comment:
As a side note (I don't believe it will be required since with the tip above we should not need to exclude apps anymore)

This here I believe can just be replaced by

```Go
if !slices.Contains(excludeApplicationsFromTestingTemplating, name) {
		return values + customLine
}
```